bindings/pyroot/src/Pythonize.cxx

```diff
@@ -2273,11 +2273,13 @@ namespace {
       Py_XDECREF(cppname);
 
       std::string pprint;
-      std::stringstream calcPrintValue;
-      calcPrintValue << "*((std::string*)" << &pprint << ") = cling::printValue((" << className << "*)"
-                     << self->GetObject() << ");";
-      gInterpreter->Calc(calcPrintValue.str().c_str());
-      return PyROOT_PyUnicode_FromString(pprint.c_str());
+      void *myObj = self->GetObject();
+      std::stringstream ss;
+      ss << myObj;
+      std::string code =  "*((" + className + "*)" + ss.str() + ")";
+      pprint = gInterpreter->Evaluate(code.c_str());
+      std::string delim = pprint.substr(pprint.find("{"));
```

Comment:
Does this fail for objects, that do not have a `{` in their string representation? E.g., `TH1F` or `TObject`? You can check out the pretty-printing unit-test!

Reply:
Ehh true